ldap_check.sh

```diff
@@ -0,0 +1,300 @@
+#       and service account password to aid in troubleshooting ldap
+#       connectivity problems.
+#
+# Questions or issues with the operation of the script, please contact
+# support@jumpcloud.com
+#
+# Author: Rob Holden | rholden@jumpcloud.com
+# 
+###############################################################################
+
+clear
+
+#
+# Help screen
+#
+
+help() {
+
+echo "###################################################################"
+echo "#                                                                 #"
+echo "#          LDAP setup and troubleshooting utility                 #"
+echo "#                                                                 #"
+echo "###################################################################"
+echo ""
+echo "Enter q to exit"
+echo ""
+echo "This utility is meant to assist in testing and configuring LDAP	"
+echo "with JumpCloud using the native ldapsearch command.            	"
+echo ""
+echo "Step 1. Enter account variables					"
+echo "	LDAP Binding User account name:  If you do not know this, please"
+echo "	refer to http://support.jumpcloud.com/knowledgebase/articles/474"
+echo "	035-configuring-your-application-with-jumpcloud-ldap		"
+echo ""
+echo "	Password:  This is the password for the LDAP Binding User	"
+echo ""
+echo "	Organization ID: if you do not know this, please refer to http:/"
+echo "	/support.jumpcloud.com/knowledgebase/articles/413926-jumpcloud-s-"
+echo "	hosted-ldap-service						"
+echo ""
+echo "Step 2. Pick a port using option 2 or 3				"
+echo ""
+echo "Step 3. Option 4 will allow you to connect to LDAP and search  	"
+echo "using the defined account variables				"
+echo ""
+echo "Option 5 will list common strings used by the applications	"
+echo "that will athenticate to JumpCloud LDAP.				"
+echo ""
+echo "Option 9 displays this help screen				"
+echo ""
+echo "Option 0 will exit the utility					"
+}
+
+
+#
+# ldapsearch
+#
+
+# Check for ldapsearch command
+
+ldapsearch=`which ldapsearch`
+
+which ldapsearch &>/dev/null;
+
+if [ $? != 0 ]
+	then
+		echo "ldapsearch is not in your path or not installed.  Please correct and rerun this script";
+		exit 1;
+fi
+
+
+# Check account vars have been entered
+
+check_ldap_config() {
+
+if [ -z "$port" ]
+        then
+                echo "Port Undefined, using default 636";
+                port=636;
+                read -rsp $'press key to continue \n' -n1 key
+fi
+
+if [ "$port" = "389 -ZZ" ]
+        then
+                uri=ldap
+        else
+                uri=ldaps
+fi
+
+if [ -z "$user" ]
+        then
+                echo -e "\nAccount variables undefined, enter information and reselect\n";
+                read_account;
+fi
+}
+
+# Define ldapsearch 
+
+ldsearch() {
+
+check_ldap_config
+
+$ldapsearch -H ${uri}://ldap.jumpcloud.com:${port} -x -b "ou=Users,o=${oid},dc=jumpcloud,dc=com" -D "uid=${user},ou=Users,o=${oid},dc=jumpcloud,dc=com" -w "${pass}" "(objectClass=${search_param})" | less
```

Comment:
If you're just going to run ldapsearch by getting it from the path location via 'which', you can just run it directly rather than using the variable.